catroid/src/main/java/org/catrobat/catroid/ui/WebViewActivity.java

```diff
@@ -337,4 +345,24 @@ private boolean isWhatsappInstalled() {
 			return false;
 		}
 	}
+
+	@Override
+	protected void onStart() {
+		super.onStart();
+		if (progressDialog != null) {
+			return;
+		}
+		progressDialog = new ProgressDialog(WebViewActivity.this);
+	}
+
+	@Override
+	protected void onDestroy() {
+		super.onDestroy();
+		if (progressDialog == null) {
+			return;
+		}
+		dismissProgressDialog();
+		progressDialog = null;
+		webView.setDownloadListener(null);
```

Comment:
IMO only setting the DownloadListener to null is sufficient at first. If the error still occurs after the fix, we need to take a look at it again. What do you think?